File: yoloq/src/main/java/com/example/yoloq/models/Reaction.java
package com.example.yoloq.models;


import com.example.yoloq.enums.ReactionType;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table
@Getter
@Setter
@NoArgsConstructor
public class Reaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private ReactionType type;

    @Column(nullable = false)
    private LocalDateTime timestamp;

    @Column(nullable = false)
    private boolean isDeleted;

    @ManyToOne(fetch = FetchType.LAZY)
    private Comment reactedTo;

    @ManyToOne
    private Post postReactedTo;

    @ManyToOne
    private User reactedBy;

}

File: client/src/app/domains/entity/Reaction.ts
import { ReactionType } from '../enums/ReactionType';
import User from './User';

export default interface Reaction {
  id: number | null;
  type: ReactionType;
  reactedBy: User;
  commentIdReactedTo: number | null;
  postIdReactedTo: number | null;
}
